src/commandHandlers/iam.ts

```diff
@@ -102,15 +150,19 @@ async function tryAddOpenSourceUserSubscription(
     const subscriptions: string[] = data != null ? data.subscriptions : [];
 
     // Add the open source subscription to the list of existing subscriptions
-    await db
-      .collection('usersSubscriptions') // TODO: create a constant variable with all these collection names (e.g. exported in firebase.ts)
-      .doc(message.author.id)
-      .set(
-        {
-          subscriptions: [...subscriptions, UserSubscriptions.OPEN_SOURCE],
-          username: message.author.username,
-        },
-        { merge: true }
-      );
+    if (toAdd) {
+      await db
+        .collection('usersSubscriptions') // TODO: create a constant variable with all these collection names (e.g. exported in firebase.ts)
+        .doc(message.author.id)
+        .set(
+          {
+            subscriptions: [...subscriptions, UserSubscriptions.OPEN_SOURCE],
+            username: message.author.username,
+          },
+          { merge: true }
+        );
+    } else {
+      await db.collection('userSubscription').doc(message.author.id).delete();
```

Comment:
Correct me if I'm wrong, but I think we don't want to delete the whole document if someone doesn't want the `opensource` role. I'm saying this because we might want other subscriptions in the future, and unsubbing from `opensource` would result in unsubbing from all of them 😅

Reply:
Oh yes. I was having doubts with regard to the usage of this.
But, then I had a little conversation with Eddie and since we're eventually moving to Mongo, perhaps we could leave this like it is until we get the API or should I edit this?

![Screenshot 2020-12-06 at 1 04 11 PM](https://user-images.githubusercontent.com/62864373/101274498-81755480-37c4-11eb-8bf6-55b5d4ae583e.png)

Reply:
@Vyvy-vi good point 👍, let's delete the whole document since supporting new subscriptions will be after the mongo / API. With the API this code will change, so it's better to keep it simple (plus it's a one-liner 😄)